aten/src/ATen/TensorIterator.cpp

```diff
@@ -343,15 +351,19 @@ void TensorIteratorBase::compute_types(const TensorIteratorConfig& config) {
     return;
   }
 
-  // Computes a common dtype, if needed
-  if (has_different_input_dtypes && config.promote_inputs_to_common_dtype_) {
-    common_dtype_ = compute_common_dtype();
-  }
-
-  // Promotes common dtype to the default float scalar type, if needed
-  if (config.promote_integer_inputs_to_float_ &&
-      c10::isIntegralType(common_dtype_, /*include_bool=*/true)) {
-    common_dtype_ = c10::typeMetaToScalarType(c10::get_default_dtype());
+  if (is_common_dtype_specified) {
```

Comment:
If we load specified_common_dtype into common_dtype_ then I don't think the two if checks in the original code need to change? The body of the first one needs to change, though.